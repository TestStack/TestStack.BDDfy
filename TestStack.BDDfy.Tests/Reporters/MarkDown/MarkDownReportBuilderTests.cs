using System;
using System.Runtime.CompilerServices;
using NUnit.Framework;
using TestStack.BDDfy.Reporters;
using TestStack.BDDfy.Reporters.MarkDown;

namespace TestStack.BDDfy.Tests.Reporters.MarkDown
{
    [TestFixture]
    public class MarkDownReportBuilderTests
    {
        [Test]
        [MethodImpl(MethodImplOptions.NoInlining)]
        public void ShouldProduceExpectedMarkdown()
        {
            Func<FileReportModel> model = () => new FileReportModel(new ReportTestData().CreateMixContainingEachTypeOfOutcome());
            var sut = new MarkDownReportBuilder();
            ReportApprover.Approve(model, sut);
        }

        [Test]
        [MethodImpl(MethodImplOptions.NoInlining)]
        public void ShouldProduceExpectedMarkdownWithExamples()
        {
            Func<FileReportModel> model = () =>
                new FileReportModel(new ReportTestData().CreateTwoStoriesEachWithOneFailingScenarioAndOnePassingScenarioWithThreeStepsOfFiveMillisecondsAndEachHasTwoExamples())
                {
                    RunDate = new DateTime(2014, 3, 25, 11, 30, 5)
                };

            var sut = new MarkDownReportBuilder();
            ReportApprover.Approve(model, sut);
        }
    }
}
