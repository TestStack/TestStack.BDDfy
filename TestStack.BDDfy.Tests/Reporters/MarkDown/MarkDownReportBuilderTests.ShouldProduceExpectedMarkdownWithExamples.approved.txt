## Story: Account holder withdraws cash
 **As an account holder**  
 **I want to withdraw cash**  
 **So that I can get money when the bank is closed**  

### Example Scenario
  Given a &lt;sign&gt; account balance  
  When the account holder requests money  
  Then money &lt;action&gt; dispensed  

### Examples: 

    | sign     | action | Result | Errors                                                                                                      |
    | positive | is     | Passed |                                                                                                             |
    | negative | is not | Failed | Step: Then money &lt;action&gt; dispensed failed with exception: [Boom With New lines] [Details at 1 below] |

#### Exceptions:
    1. Boom With New lines
       at TestStack.BDDfy.Tests.Reporters.ReportTestData.GetScenarios(Boolean includeFailingScenario, Boolean includeExamples) in ...\ReportTestData.cs:line 114

## Story: Happiness
 **As a person**  
 **I want ice cream**  
 **So that I can be happy**  

### Example Scenario
  Given a &lt;sign&gt; account balance  
  When the account holder requests money  
  Then money &lt;action&gt; dispensed  

### Examples: 

    | sign     | action |
    | positive | is     |
    | negative | is not |
